test/e2e/test.js

```diff
@@ -35,118 +34,169 @@ var SCOPES = [
+
+        var promises = body.breakpoints.map(function(breakpoint) {
+          return deleteBreakpoint(debuggee, breakpoint);
+        });
+
+        var deleteAll = Q.all(promises);
+
+        return deleteAll;
+      })
+      .then(function(deleteResults) {
+        console.log('-- delete results', deleteResults);
+        return debuggee;
+      })
+      .then(function(debuggee) {
+        // Set a breakpoint
+        console.log('-- setting a breakpoint');
+        var promise = setBreakpoint(debuggee, {
+          id: 'breakpoint-1',
+          location: {path: 'test.js', line: 5},
+          condition: 'n === 10'
+        });
+        // I don't know what I am doing. There is a better way to write the
+        // following using promises.
+        var result = promise.then(function(body) {
+          console.log('-- resolution of setBreakpoint', body);
+          assert.ok(body.breakpoint, 'should have set a breakpoint');
+          var breakpoint = body.breakpoint;
+          assert.ok(breakpoint.id, 'breakpoint should have an id');
+          assert.ok(breakpoint.location, 'breakpoint should have a location');
+          assert.strictEqual(breakpoint.location.path, 'test.js');
+          return { debuggee: debuggee, breakpoint: breakpoint };
+        });
+        return result;
+      })
+      .then(function(result) {
+        assert.ok(result.breakpoint);
+        assert.ok(result.debuggee);
+        console.log('-- waiting a bit before running fib');
+        return Q.delay(20 * 1000).then(function() { return result; });
+      })
+      .then(function(result) {
+        console.log('-- Running fib');
+        fib(12);
+        console.log('-- waiting before checking if the breakpoint was hit');
+        return Q.delay(10 * 1000).then(function() { return result; });
+      })
+      .then(function(result) {
+        console.log('-- now checking if the breakpoint was hit');
+        var promise = getBreakpoint(result.debuggee, result.breakpoint);
+        return promise;
+      })
+      .then(function(body) {
+        console.log('-- results of get breakpoint\n', body);
+        assert.ok(body.breakpoint, 'should have a breakpoint in the response');
+        var hit = body.breakpoint;
+        assert.ok(hit.isFinalState, 'breakpoint should have been hit');
+        assert.ok(Array.isArray(hit.stackFrames), 'should have stack ');
+        var top = hit.stackFrames[0];
+        assert.ok(top, 'should have a top entry');
+        assert.ok(top.function, 'frame should have a function property');
+        assert.strictEqual(top.function, 'fib');
+
+        var arg = _.find(top.arguments, function(a) {
+          return a.name === 'n';
+        });
+        assert.ok(arg, 'should find the n argument');
+        assert.strictEqual(arg.value, '10');
+        console.log('Test passed');
+        return true;
+      })
+      .catch(function(err) {
+        console.error(err);
+        process.exit(2);
```

Comment:
I can let it throw out of the promise. I was originally contemplating doing something with the error, but this doesn't seem necessary anymore.